### PR TITLE
knapsack: add exercise

### PR DESCRIPTION
**Pull request by colinleach**
The linter is going to complain. Too bad, it's a long URL inside a comment, so live with it.

More seriously, I'm not thrilled with the `example.R`. I had hoped to vectorize this, but each cell update needs access to the cells above and to the left. Hence needing both matrix indices, hence nested loops. If you have a better idea, please change the file and add yourself as an author.

I've put difficulty 5, copying Python, Julia, C#.

### CODE REVIEW

File: exercises/practice/knapsack/.meta/example.R [hunk 1]
```diff
@@ -0,0 +1,28 @@
+maximum_value <- function(max_wt, items) {
+  # This URL is > 80 chars, ignore whining from the linter:
+  # https://en.wikipedia.org/wiki/Knapsack_problem#Dynamic_programming_in-advance_algorithm
```
Comment (ErikSchierboom) [inline exercises/practice/knapsack/.meta/example.R:3]:
```suggestion
  # nolint start
  # https://en.wikipedia.org/wiki/Knapsack_problem#Dynamic_programming_in-advance_algorithm
  # nolint end
```

Comment (ErikSchierboom) [reply exercises/practice/knapsack/.meta/example.R:3]: This fixes the linting problem

File: exercises/practice/knapsack/knapsack.R [hunk 2]
```diff
@@ -0,0 +1,3 @@
+maximum_value <- function(max_wt, items) {
```
Comment (ErikSchierboom) [inline exercises/practice/knapsack/knapsack.R:1]: Maybe make this slightly more verbose?

```suggestion
maximum_value <- function(max_weight, items) {
```

**Comment by colinleach** [comment]
Thanks for the linter advice. I failed to RTFM!

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**